pkg/hostpath/controllerserver.go

```diff
@@ -84,6 +90,23 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		glog.V(3).Infof("failed to create volume: %v", err)
 		return nil, err
 	}
+	if req.GetVolumeContentSource() != nil {
+		contentSource := req.GetVolumeContentSource()
+		if contentSource.GetSnapshot() != nil {
+			snapshotId := contentSource.GetSnapshot().GetId()
+			snapshot, ok := hostPathVolumeSnapshots[snapshotId]
```

Comment:
We keep them in memory? So if we restart the driver they get lost, right? if this is is true, we should document this somewhere. I am ok with having it memory and documenting that this is a test-only driver.